src/libraries/System.Net.Primitives/src/System/Net/IPNetwork.cs

```diff
@@ -0,0 +1,400 @@
+        {
+            if (!TryParseCore(s, out IPAddress? address, out int prefixLength))
+            {
+                throw new FormatException(SR.net_bad_ip_network);
+            }
+
+            try
+            {
+                return new IPNetwork(address, prefixLength);
+            }
+            catch (Exception inner)
+            {
+                throw new FormatException(inner.Message, inner);
+            }
+        }
+
+        /// <summary>
+        /// Converts the specified CIDR string to an <see cref="IPNetwork"/> instance and returns a value indicating whether the conversion succeeded.
+        /// </summary>
+        /// <param name="s">A <see cref="string"/> that defines an IP network in CIDR notation.</param>
+        /// <param name="result">When the method returns, contains an <see cref="IPNetwork"/> instance if the conversion succeeds.</param>
+        /// <returns><see langword="true"/> if the conversion was succesful; otherwise, <see langword="false"/>.</returns>
+        public static bool TryParse(string? s, out IPNetwork result)
+        {
+            if (s == null)
+            {
+                result = default;
+                return false;
+            }
+
+            return TryParse(s.AsSpan(), out result);
+        }
+
+        /// <summary>
+        /// Converts the specified CIDR character span to an <see cref="IPNetwork"/> instance and returns a value indicating whether the conversion succeeded.
+        /// </summary>
+        /// <param name="s">A <see cref="string"/> that defines an IP network in CIDR notation.</param>
+        /// <param name="result">When the method returns, contains an <see cref="IPNetwork"/> instance if the conversion succeeds.</param>
+        /// <returns><see langword="true"/> if the conversion was succesful; otherwise, <see langword="false"/>.</returns>
+        public static bool TryParse(ReadOnlySpan<char> s, out IPNetwork result)
+        {
+            if (TryParseCore(s, out IPAddress? address, out int prefixLength) && Validate(address, prefixLength, throwOnFailure: false))
+            {
+                result = new IPNetwork(address, prefixLength, false);
+                return true;
+            }
+
+            result = default;
+            return false;
+        }
+
+        private static bool TryParseCore(ReadOnlySpan<char> s, [NotNullWhen(true)] out IPAddress? address, out int prefixLength)
+        {
+            const int MaxCharsCountAfterIpAddress = 4;
+            const int MinCharsRequired = 4;
+
+            if (s.Length < MinCharsRequired)
+            {
+                goto Failure;
+            }
+
+            int separatorExpectedStartingIndex = s.Length - MaxCharsCountAfterIpAddress;
+            int separatorIndex = s
+                .Slice(separatorExpectedStartingIndex)
+                .IndexOf('/');
+
+            if (separatorIndex != -1)
+            {
+                separatorIndex += separatorExpectedStartingIndex;
```

Comment:
Can this be simplified to something like
```suggestion
            int separatorIndex = s.LastIndexOf('/');
            if (separatorIndex >= 0)
            {
```
?